src/pages/Defi/components/OpportunityCard.tsx

```diff
@@ -40,38 +46,51 @@ export const OpportunityCard = ({
   cryptoAmount,
   fiatAmount,
   expired,
+  moniker,
 }: OpportunityCardProps) => {
   const history = useHistory()
   const location = useLocation()
   const bgHover = useColorModeValue('gray.100', 'gray.700')
-  const network = NetworkTypes.MAINNET
-  const assetNamespace = AssetNamespace.ERC20
+  const network = chainTypeToMainNetNetworkType(chain)
```

Comment:
If using a `chainId` instead of a `chain`, we should be able to surface the correct network and asset namespace using `caip2.fromCaip2` without an additional util.

Reply:
Fixed, I went ahead and removed that way of getting the assetCaip19, we have assetId available in the props of this component ... It seems well constructed in every use case for `OpportunityCard`, I just wasnt able to verify the cards are working for yearn / fox opportunities, but it should.